gcp_variant_transforms/testing/integration/run_preprocessor_tests.py

```diff
@@ -0,0 +1,178 @@
+
+    self._pipelines_api_request = {
+        'pipelineArgs': {
+            'projectId': context.project,
+            'logging': {'gcsPath': context.logging_location},
+            'serviceAccount': {'scopes': _SCOPES}
+        },
+        'ephemeralPipeline': {
+            'projectId': context.project,
+            'name': _PIPELINE_NAME,
+            'resources': {'zones': _DEFAULT_ZONES},
+            'docker': {
+                'imageName': context.image,
+                'cmd': ' '.join([_SCRIPT_PATH] + args)
+            }
+        }
+    }
+
+  def validate_result(self):
+    """Validates the results.
+
+    - Checks that the report is generated.
+    - Validates report's contents are the same as ``expected_contents``.
+    """
+    if not filesystems.FileSystems.exists(self._report_path):
+      raise run_tests_common.TestCaseFailure(
+          'Report is not generated in {}'.format(self._report_path))
+    with filesystems.FileSystems.open(self._report_path) as f:
+      for expected_content in self._expected_contents:
+        line = f.readline()
+        if expected_content != line:
+          raise run_tests_common.TestCaseFailure(
+              'Contents mismatch: expected {}, got {}'.format(
+                  expected_content, line))
+    if not self._keep_reports:
+      filesystems.FileSystems.delete(self._report_path)
+
+
+class TestContextManager(object):
+  """Manages all resources for a given run of tests."""
+
+  def __init__(self, args):
+    # type: (argparse.ArgumentParser) -> None
+    self.staging_location = args.staging_location
+    self.temp_location = args.temp_location
+    self.logging_location = args.logging_location
+    self.project = args.project
+    self.credentials = GoogleCredentials.get_application_default()
+    self.image = args.image
+    self.keep_reports = args.keep_reports
+
+  def __enter__(self):
+    return self
+
+  def __exit__(self, *args):
+    return
+
+
+def _get_args():
+  parser = argparse.ArgumentParser()
+  run_tests_common.add_args(parser)
+  parser.add_argument('--keep_reports',
+                      type=bool, default=False, nargs='?', const=True,
+                      help='If set, generated reports and resolved headers are '
+                           'not deleted.')
+  return parser.parse_args()
+
+
+def _get_test_configs():
+  # type: () -> List[Dict]
+  """Gets all test configs in preprocessor_tests."""
+  required_keys = ['test_name', 'report_path', 'input_pattern',
+                   'expected_contents']
+  test_file_path = os.path.join(os.getcwd(), _TEST_FOLDER)
+  return run_tests_common.get_configs(test_file_path, required_keys)
+
+
+def _run_test(test, context):
+  test.run(context)
+
+
+def main():
+  """Runs the integration tests for preprocessor."""
+  args = _get_args()
+  test_case_configs = _get_test_configs()
+  with TestContextManager(args) as context:
+    pool = multiprocessing.Pool(processes=len(test_case_configs))
```

Comment:
A general comment/discussion since now we are extending this `multiprocessing` pattern even more: I inherited this pattern from the first implementation but at some point I was thinking that we don't really need multiprocessing here. The tests can be submitted to the Pipelines API (which is non-blocking) and then the operations being monitored in a loop. IOW instead of having individual blocking `_wait_for_operation_done` for each test, there would be one `_wait_for_all_operation_done` that checks all operations in a loop.

Do you see any reason for not using this non-multiprocessing pattern? The problem with the multiprocessing is that it adds unnecessary complexity both in the code and when running it. If you also don't see any reason, then please add a TODO or create a GitHub issue to remove this `multiprocessing` pattern from the other test module (in a future PR) and avoid it for this new module in this PR. Maybe add one extra test case (to have two instead of one) as well to make sure you have tested this different approach as well.